### PR TITLE
Alias feature

### PR DESCRIPTION
**Pull request by gfiorav**
## Context

This PR contains the code base for the alias feature.

## Acceptance

It's recommended that you test:
+ [ ] Add table alias
+ [ ] Remove table alias
+ [ ] Add column alias
+ [ ] Remove table alias
+ [ ] Other normal column operations
+ [ ] Import an non-aliased dataset from Data Library
+ [ ] Import an aliased dataset from Data Library

### CODE REVIEW

**Comment by gfiorav** [comment]
Ready for you to accept and merge, @mbektas

File: lib/assets/javascripts/cartodb/table/table.js [hunk 1]
```diff
@@ -332,12 +345,16 @@ $(function() {
       this.mapTab.bind('mapViewReady', this.header.addOverlaysDropdown, this.header);
       this.mapTab.bind('missingClick', self.menu.hide, self.menu);
 
+      this.titleDropdown.bind('renameAlais', this._saveTitleAlias, this);
```
Comment (mbektash) [inline lib/assets/javascripts/cartodb/table/table.js]: is this typo/important? renameAlais

Comment (gfiorav) [reply lib/assets/javascripts/cartodb/table/table.js]: Wow, nice catch. Makes me wonder how it works. I'll review!

File: lib/carto/http_header_authentication.rb [hunk 2]
```diff
@@ -85,8 +85,14 @@ def field_from_value(request)
 
     def header_value(headers)
       header = ::Cartodb.get_config(:http_header_authentication, 'header')
-      puts "user-auto-creation : Trying to extract value from headers for #{header}, value is #{headers[header]}"
-      !header.nil? && !header.empty? ? headers[header] : nil
+
```
Comment (mbektash) [inline lib/carto/http_header_authentication.rb]: This change doesnt seem to be related to aliases?

File: spec/lib/carto/http_header_authentication_spec.rb [hunk 3]
```diff
@@ -43,6 +43,12 @@
       stub_http_header_authentication_configuration(field: 'email')
       Carto::HttpHeaderAuthentication.new.valid?(mock_email_request).should == true
     end
+
```
Comment (mbektash) [inline spec/lib/carto/http_header_authentication_spec.rb]: This change doesnt seem to be related to aliases?

File: lib/assets/javascripts/cartodb/table/table.js (hunk 1 above)
Comment (gfiorav) [reply lib/assets/javascripts/cartodb/table/table.js]: It's replicated on the trigger: https://github.com/cartodb-org/cartodb/blob/aliases/lib/assets/javascripts/cartodb/table/table_title_dropdown.js#L60

Fixing and pushing

File: lib/carto/http_header_authentication.rb (hunk 2 above)
Comment (gfiorav) [reply lib/carto/http_header_authentication.rb]: Sorry, I'll check this file out.

File: spec/lib/carto/http_header_authentication_spec.rb (hunk 3 above)
Comment (gfiorav) [reply spec/lib/carto/http_header_authentication_spec.rb]: Sorry, I'll check this file out.

**Comment by gfiorav** [comment]
I removed the unrelated code, sorry about that. Please continue @mbektas !

**Comment by mbektash** [comment]
ok, thanks

**Pull request merged**